x-pack/plugins/cases/server/client/cases/update_custom_field.ts

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Boom from '@hapi/boom';
+
+import type { CasesClient, CasesClientArgs } from '..';
+
+import type { CaseRequestCustomFields, CustomFieldPatchRequest } from '../../../common/types/api';
+import { CustomFieldPatchRequestRt } from '../../../common/types/api';
+import { Operations } from '../../authorization';
+import { createCaseError } from '../../common/error';
+import { flattenCaseSavedObject } from '../../common/utils';
+import { decodeOrThrow } from '../../../common/api/runtime_types';
+import type { Cases } from '../../../common/types/domain';
+import { decodeWithExcessOrThrow } from '../../../common/api';
+import { CasesRt } from '../../../common/types/domain';
+import {
+  validateCustomFieldKeysAgainstConfiguration,
+  validateCustomFieldTypesInRequest,
+} from './validators';
+import type { UserActionEvent } from '../../services/user_actions/types';
+
+export interface UpdateCustomFieldArgs {
+  /**
+   * The ID of a case
+   */
+  caseId: string;
+  /**
+   * The ID of an custom field to be updated
+   */
+  customFieldId: string;
+  /**
+   * details of custom field to update
+   */
+  customFieldPatchDetails: CustomFieldPatchRequest;
+}
+
+/**
+ * Updates the specified cases with new values
+ *
+ * @ignore
+ */
+export const updateCustomField = async (
+  args: UpdateCustomFieldArgs,
+  clientArgs: CasesClientArgs,
+  casesClient: CasesClient
+): Promise<Cases> => {
+  const {
+    services: { caseService, userActionService },
+    user,
+    logger,
+    authorization,
+  } = clientArgs;
+
+  try {
+    const { customFieldPatchDetails, customFieldId, caseId } = args;
+    const { customFieldDetails, version } =
+      decodeWithExcessOrThrow(CustomFieldPatchRequestRt)(customFieldPatchDetails);
+    const updatedAt = new Date().toISOString();
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const { username, full_name, email, profile_uid } = user;
+
+    const caseToUpdate = await caseService.getCase({
+      id: caseId,
+    });
+
+    await authorization.ensureAuthorized({
+      entities: [{ owner: caseToUpdate.attributes.owner, id: caseToUpdate.id }],
+      operation: Operations.updateCase,
+    });
+
+    if (customFieldDetails.value == null) {
+      throw Boom.badRequest('Custom field value cannot be null or undefined.');
+    }
+
+    const configurations = await casesClient.configure.get({
+      owner: caseToUpdate.attributes.owner,
+    });
+
+    const customField: CaseRequestCustomFields = [{ key: customFieldId, ...customFieldDetails }];
+
+    const customFieldIndex = caseToUpdate.attributes.customFields.findIndex(
```

Comment:
I had this check but `validateCustomFieldKeysAgainstConfiguration ` also verifies it, so removed it

Reply:
Good point. It might be covered by `decodeWithExcessOrThrow` as we want to check it against
```
// we use the schema of the patch case request that will do the value schema validation for us
 const decodedCustomField = decodeWithExcessOrThrow(CustomFieldsRt)([{ value, type: foundCustomField.type, key: customFieldId, // ...etc }]);

```
But still a check whether we found customField or not would be more clear 👍

Reply:
I don't think it will get covered by the `CustomFieldsRt` because `decodeWithExcessOrThrow(CustomFieldsRt)([]) // empty array` is valid!